app/docs/src/reference/authentication.md

```diff
@@ -0,0 +1,107 @@
+# Authentication Functions
+
+Authentication functions are a kind of [function](./function.md) which handle
+credential validation and setup before other functions execute. They run
+automatically when a function needs to use [secrets](./secrets.md), ensuring
+that authentication happens transparently and consistently.
+
+## How Authentication Functions Work
+
+Authentication functions are defined on assets that use secrets (configured
+using the SecretDefinitionBuilder API). When any other function needs to use a
+secret:
+
+1. System Initiative identifies which secret is required
+2. The authentication function for that secret runs first
+3. The authentication function sets up credentials (environment variables or
+   request storage)
+4. The original function executes with access to the authenticated session
+
+This ensures that credentials are always properly configured before any API
+calls or operations that require authentication.
+
+## When Authentication Functions Run
+
+Authentication functions execute automatically:
+
+- Before any action function that uses a secret
+- Before any attribute function that requires authentication
+- Once per request, even if multiple functions use the same secret
+
+You never need to manually invoke authentication functions - System Initiative
+handles this automatically based on function dependencies.
+
+## What Authentication Functions Do
+
+Authentication functions typically:
+
+- **Set environment variables**: Configure credentials like `AWS_ACCESS_KEY_ID`
+  or `DOCKER_AUTH_CONFIG` that external tools expect
+- **Store session data**: Use the requestStorage API to pass authentication data
+  between functions
+- **Handle multiple authentication methods**: Support different authentication
+  flows (API keys, assume role, OAuth, etc.)
+- **Prepare configuration files**: Write credential files that command-line
+  tools require (like Docker config or kubeconfig)
```

Comment:
I don't think we can do this since we do not save it on disk... unless I am missing something.

Worth double checking our public and beta credentials to see what their auth func bodies have.